### PR TITLE
ci(ci): Enable self hosted checks for forks

### PR DESCRIPTION
**Pull request by dav1d**
#skip-changelog

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (hubertdeng123) [inline .github/workflows/ci.yml:1]: The `Build and push to ghcr.io` in `build-docker` job is being skipped due to being a fork, We'll need to change this too if we 'd like to enable self-hosted checks for forks

Comment (Dav1dde) [reply .github/workflows/ci.yml:1]: The job now imports the image from a build artifact if access to ghcr is not available.

**Comment by Dav1dde** [comment]
Mh still doesn't work because the install script also tries to pull https://github.com/getsentry/self-hosted/blob/c40b1530d104e9481d172e06dc6810bea0795d5c/install/ensure-relay-credentials.sh#L27

**Comment by Dav1dde** [comment]
We don't want to update the self hosted install scripts to accommodate for this.

**Pull request closed without merging**